UEFITool NE 59

vampirecat35 opened the pull request:
- Updates for QT 5 and QT 6
- Deprecation fixes
- Code fixes

vit9696 commented:
Hello, please update the compatibility with the CI. We still require Qt 5.6 as a minimum for LGPL 2.0 license reasons.

vampirecat35 commented:
> Hello, please update the compatibility with the CI. We still require Qt 5.6 as a minimum for LGPL 2.0 license reasons.

I used Qt-5.15.2 and Qt-6.0.3
I tested it on macOS.

vampirecat35 commented:
> Hello, please update the compatibility with the CI. We still require Qt 5.6 as a minimum for LGPL 2.0 license reasons.

Done.
I added a QT version check.
If 5.11 or older use .width
Otherwise use new method.

vit9696 commented on .github/workflows/main.yml:
```diff
@@ -1,4 +1,4 @@
-name: CI
+x@name: CI
```
You broke CI completely here by accident.

vampirecat35 replied:
Fixed.
Accidentally edited the file.

vit9696 commented on UEFITool/uefitool.pro:
```diff
@@ -8,6 +8,8 @@ DEFINES += "U_ENABLE_NVRAM_PARSING_SUPPORT"
 DEFINES += "U_ENABLE_ME_PARSING_SUPPORT"
 DEFINES += "U_ENABLE_GUID_DATABASE_SUPPORT"
 
+greaterThan(QT_MAJOR_VERSION, 5): DEFINES += "QT_VERSION_MAJOR=6"
```
Are you sure there is no builtin macro to check against Qt version?

vampirecat35 replied:
True.
This isn't needed.
Pushed commit

vampirecat35 commented:
CI is fixed and build success.
Also tested on newer QT versions and success too.

vit9696 commented on UEFITool/ffsfinder.cpp:
```diff
@@ -29,7 +29,12 @@ USTATUS FfsFinder::findHexPattern(const UModelIndex & index, const UByteArray &
 
     bool hasChildren = (model->rowCount(index) > 0);
     for (int i = 0; i < model->rowCount(index); i++) {
+#ifdef _USE_DEPRECATED
```
Why cannot we use non-deprecated interfaces exclusively? Does Qt 5.6 not support them? Who does define `_USE_DEPRECATED`?

vampirecat35 replied:
I'll do a version check.
Fixing in a minute

vampirecat35 commented:
Fixed older QT versions.
Newer ones will use the new code.

vit9696 commented on common/treemodel.h:
```diff
@@ -86,7 +88,7 @@ class UModelIndex
 class TreeModel : public QAbstractItemModel
 {
-    Q_OBJECT
+    //Q_OBJECT
```
Better just remove than comment out.

vampirecat35 replied:
Removed

vit9696 commented on common/treemodel.h:
```diff
@@ -35,7 +35,9 @@ enum ItemFixedState {
 #include "types.h"
 #include "treeitem.h"
 
+#ifndef UModelIndex
```
Why is that one needed?

vampirecat35 replied:
Removing...

vit9696 commented on common/ustring.cpp:
```diff
@@ -19,7 +19,7 @@ UString usprintf(const char* fmt, ...)
     va_list vl;
     va_start(vl, fmt);
-    msg.vsprintf(fmt, vl);
+    msg = msg.vasprintf(fmt, vl);
```
Why is this necessary?

vampirecat35 replied:
Newer QT versions like QT 6 don't implement .vsprintf.
This thus needs the newer method.

vit9696 commented:
Looks mostly good, spotted a few more weird points, but should be ready to merge soon after.

vampirecat35 commented:
I added it using a version check.
This way older than QT 5.6 also builds.

vampirecat35 commented:
Image of UEFITool on macOS Big Sur using QT 6
[https://i.ibb.co/WnFR9YZ/UEFITool-NE-59-mac-OS-QT6.png](UEFITool QT6)

vampirecat35 commented:
Removed Xcode projects, copied new icon file, added uefitool.entitlements for use with code sign
codesign -fs "Apple Development" --entitlements uefitool.entitlements was used in my case.
This is needed after macdeployqt for adding sandbox and signing.

vampirecat35 commented:
Done with all QT 6 updates.
It now builds without a warning for QT 6.
